fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -2373,14 +2380,21 @@ def alloc_storage_(data: torch.Tensor, size: torch.Size) -> None:
 
 
 def _post_state_dict_hook(
-    module: FullyShardedDataParallel, state_dict: "OrderedDict[str, torch.Tensor]", prefix: str, *args: Any
+    state_dict_on_rank_0_only: bool,
```

Comment:
can we move this after prefix string so that we add to it after current args?

Reply:
I don't think so.. 
functools.partial(func_x, a) is creating a new function func_y which doesn't need param a, that's what we need base on how the hook defined here https://github.com/pytorch/pytorch/blob/0460324b9bf7182dde9fa17bd498f0440e41c345/torch/nn/modules/module.py#L1246-L1255, which will take a method as input and the params pass into the method is positional argument *arg.
If we move the state_dict_on_rank_0_only to the last and define `self._register_state_dict_hook(functools.partial(_post_state_dict_hook, state_dict_on_rank_0_only=self.state_dict_on_rank_0_only))` we will get ` func() got multiple values for keyword argument state_dict_on_rank_0_only` error